src/Microsoft.AspNetCore.Mvc.Core/Internal/FileContentResultExecutor.cs

```diff
@@ -15,15 +17,51 @@ public FileContentResultExecutor(ILoggerFactory loggerFactory)
 
         public Task ExecuteAsync(ActionContext context, FileContentResult result)
         {
-            SetHeadersAndLog(context, result);
-            return WriteFileAsync(context, result);
+            var rangeInfo = new(RangeItemHeaderValue range, long rangeLength)?();
+            if (result.LastModified.HasValue)
+            {
+                rangeInfo = SetHeadersAndLog(
+                    context,
+                    result,
+                    result.FileContents.Length,
+                    result.LastModified.Value,
+                    result.EntityTag);
+            }
+            else
+            {
+                rangeInfo = SetHeadersAndLog(
+                    context,
+                    result,
+                    result.FileContents.Length);
+            }
+
+            if (rangeInfo.HasValue)
+            {
+                return WriteFileAsync(context, result, rangeInfo.Value.range, rangeInfo.Value.rangeLength);
+            }
+
+            return WriteFileAsync(context, result, null, 0);
         }
 
-        private static Task WriteFileAsync(ActionContext context, FileContentResult result)
+        private static Task WriteFileAsync(ActionContext context, FileContentResult result, RangeItemHeaderValue range, long rangeLength)
         {
             var response = context.HttpContext.Response;
+            var outputStream = response.Body;
+
+            if (range == null)
+            {
+                return response.Body.WriteAsync(result.FileContents, offset: 0, count: result.FileContents.Length);
+            }
+
+            else if (rangeLength == 0)
+            {
+                return Task.CompletedTask;
+            }
 
-            return response.Body.WriteAsync(result.FileContents, offset: 0, count: result.FileContents.Length);
+            else
+            {
+                return response.Body.WriteAsync(result.FileContents, offset: (int)range.From.Value, count: (int)rangeLength);
```

Comment:
> A client can limit the number of bytes requested without knowing the
   size of the selected representation.  If the last-byte-pos value is
   absent, or if the value is greater than or equal to the current
   length of the representation data, the byte range is interpreted as
   the remainder of the representation (i.e., the server replaces the
   value of last-byte-pos with a value that is one less than the current
   length of the selected representation).

From [this](https://tools.ietf.org/html/rfc7233#section-2.1) which we do [here](https://github.com/aspnet/StaticFiles/blob/dev/shared/Microsoft.AspNetCore.RangeHelper.Sources/RangeHelper.cs#L132) in the `NormalizeRange` method in the `RangeHelper`